internal/pkg/v2/handler/core/data/mocks/device.go

```diff
@@ -0,0 +1,75 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package mocks
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/edgexfoundry/edgex-go/internal/mocks"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/types"
+	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+
+	"github.com/globalsign/mgo/bson"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/mock"
+)
+
+const testDeviceName string = "Test Device"
+
+// NewMockDeviceClient creates a new mock DeviceClient which has some general mocking behavior defined.
+func NewMockDeviceClient() *mocks.DeviceClient {
+	client := &mocks.DeviceClient{}
+
+	protocols := getProtocols()
+
+	mockDeviceResultFn := func(ctx context.Context, id string) contract.Device {
+		if bson.IsObjectIdHex(id) {
+			return contract.Device{Id: id, Name: testDeviceName, Protocols: protocols}
+		}
+		return contract.Device{}
+	}
+	client.On("Device", context.Background(), "valid").Return(mockDeviceResultFn, nil)
+	client.On("Device", context.Background(), "404").Return(mockDeviceResultFn,
+		types.NewErrServiceClient(http.StatusNotFound, []byte{}))
+	client.On("Device", context.Background(), mock.Anything).Return(mockDeviceResultFn, fmt.Errorf("some error"))
+
+	mockDeviceForNameResultFn := func(ctx context.Context, name string) contract.Device {
+		device := contract.Device{Id: uuid.New().String(), Name: name, Protocols: protocols}
+
+		return device
+	}
+	client.On("DeviceForName", context.Background(), testDeviceName).Return(mockDeviceForNameResultFn, nil)
+	client.On("DeviceForName", context.Background(), "404").Return(mockDeviceForNameResultFn,
+		types.NewErrServiceClient(http.StatusNotFound, []byte{}))
+	client.On("DeviceForName", context.Background(), mock.Anything).Return(mockDeviceForNameResultFn,
+		fmt.Errorf("some error"))
+
+	return client
+}
+
+func getProtocols() map[string]contract.ProtocolProperties {
+	p1 := make(map[string]string)
+	p1["host"] = "localhost"
+	p1["port"] = "1234"
+	p1["unitID"] = "1"
+
+	p2 := make(map[string]string)
+	p2["serialPort"] = "/dev/USB0"
+	p2["baudRate"] = "19200"
+	p2["dataBits"] = "8"
+	p2["stopBits"] = "1"
+	p2["parity"] = "0"
+	p2["unitID"] = "2"
+
+	wrap := make(map[string]contract.ProtocolProperties)
+	wrap["modbus-ip"] = p1
+	wrap["modbus-rtu"] = p2
+
+	return wrap
+}
```

Comment:
Copy from `internal/core/data/mocks/device.go` for `v2/handler/core/data/event_test.go`

Reply:
write the comment to state we will modify it once core-metadata V2 Device API is available

Reply:
adde.